src/ucp/rndv/proto_rndv.c

```diff
@@ -655,6 +655,8 @@ ucp_proto_rndv_handle_rtr(void *arg, void *data, size_t length, unsigned flags)
         /* RTR covers the whole send request - use the send request directly */
         ucs_assert(rtr->offset == 0);
 
+        ucp_datatype_iter_mem_dereg(worker->context, &req->send.state.dt_iter,
```

Comment:
is it needed because now dt_iter does not contain direct md_map?
maybe assert in ucp_datatype_iter_mem_reg that when it's called the existing memh is NULL?